src/providers/mssql/qgsmssqlprovider.cpp

```diff
@@ -1934,3 +1960,324 @@ QGISEXTERN QgsVectorLayerImport::ImportError createEmptyLayer(
+          .arg( QgsMssqlProvider::quotedValue( dsUri.table() ) )
+          .arg( QgsMssqlProvider::quotedValue( dsUri.geometryColumn() ) )
+          .arg( QgsMssqlProvider::quotedValue( styleName.isEmpty() ? dsUri.table() : styleName ) );
+  }
+  if ( useAsDefault )
+  {
+    QString removeDefaultSql = QString( "UPDATE layer_styles "
+                                        " SET useAsDefault=0"
+                                        " WHERE f_table_catalog=%1"
+                                        " AND f_table_schema=%2"
+                                        " AND f_table_name=%3"
+                                        " AND f_geometry_column=%4" )
+                               .arg( QgsMssqlProvider::quotedValue( dsUri.database() ) )
+                               .arg( QgsMssqlProvider::quotedValue( dsUri.schema() ) )
+                               .arg( QgsMssqlProvider::quotedValue( dsUri.table() ) )
+                               .arg( QgsMssqlProvider::quotedValue( dsUri.geometryColumn() ) );
+    sql = QString( "%1; %2;" ).arg( removeDefaultSql, sql );
+  }
+
+  QgsDebugMsg("Inserting styles");
+  QgsDebugMsg(sql);
+  bool execOk = query.exec( sql );
+
+  if ( !execOk )
+  {
+    errCause = QObject::tr( "Unable to save layer style. It's not possible to insert a new record into the style table. Maybe this is due to table permissions. Please contact your database administrator." );
+  }
+  return execOk;
+}
+
+
+QGISEXTERN QString loadStyle( const QString& uri, QString& errCause )
+{
+  QString style = "";
+  QgsDataSourceURI dsUri( uri );
+  // connect to database
+  QSqlDatabase mDatabase = QgsMssqlProvider::GetDatabase( dsUri.service(), dsUri.host(), dsUri.database(), dsUri.username(), dsUri.password() );
+
+  if ( !QgsMssqlProvider::OpenDatabase( mDatabase ) )
+  {
+    QgsDebugMsg( "Error connecting to database" );
+    QgsDebugMsg( mDatabase.lastError().text() );
+    return "";
+  }
+
+  QSqlQuery query = QSqlQuery( mDatabase );
+  query.setForwardOnly( true );
+
+  QString selectQmlQuery = QString( "SELECT top 1 styleQML"
+                                    " FROM layer_styles"
+                                    " WHERE f_table_catalog=%1"
+                                    " AND f_table_schema=%2"
+                                    " AND f_table_name=%3"
+                                    " AND f_geometry_column=%4"
+                                    " ORDER BY useAsDefault desc" )
+                           .arg( QgsMssqlProvider::quotedValue( dsUri.database() ) )
+                           .arg( QgsMssqlProvider::quotedValue( dsUri.schema() ) )
+                           .arg( QgsMssqlProvider::quotedValue( dsUri.table() ) )
+                           .arg( QgsMssqlProvider::quotedValue( dsUri.geometryColumn() ) );
+
+   if ( !query.exec( selectQmlQuery ) )
+   {
+     errCause = query.lastError().text();
+     QgsDebugMsg( errCause );
```

Comment:
Or Nyall can review it ;)

On Thu, May 26, 2016 at 7:49 PM, Nyall Dawson <notifications@github.com>
wrote:

> In src/providers/mssql/qgsmssqlprovider.cpp
> <https://github.com/qgis/QGIS/pull/3107#discussion_r64719020>:
>
> > +  QString selectQmlQuery = QString( "SELECT top 1 styleQML"
> > +                                    " FROM layer_styles"
> > +                                    " WHERE f_table_catalog=%1"
> > +                                    " AND f_table_schema=%2"
> > +                                    " AND f_table_name=%3"
> > +                                    " AND f_geometry_column=%4"
> > +                                    " ORDER BY useAsDefault desc" )
> > +                           .arg( QgsMssqlProvider::quotedValue( dsUri.database() ) )
> > +                           .arg( QgsMssqlProvider::quotedValue( dsUri.schema() ) )
> > +                           .arg( QgsMssqlProvider::quotedValue( dsUri.table() ) )
> > +                           .arg( QgsMssqlProvider::quotedValue( dsUri.geometryColumn() ) );
> > +
> > +   if ( !query.exec( selectQmlQuery ) )
> > +   {
> > +     errCause = query.lastError().text();
> > +     QgsDebugMsg( errCause );
>
> Change to:
>
> QgsDebugMsg( query.lastError().text();
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly or view it on GitHub
> <https://github.com/qgis/QGIS/pull/3107/files/e94ccf8a94e27c212609edfc58d50ace96433fc2#r64719020>
>